### DISCUSSION TITLE
API Key Authentication & Generating API Key

### DISCUSSION THREAD

**Discussion by countrymusicfy** [AzuraCast/AzuraCast#4531] [Q&A]
@Vaalyn Regarding the new security fix, is there any documentation on API key authentication and how to generate an API key?

------------------------------------------

API calls will now either require API key authentication or both a current active login session and a unique identifier; if you're calling the API externally, you should always use a generated API key and not count on the user's existing session.

**Comment by BusterNeece** [comment]
@countrymusicfy This shouldn't affect how you've consumed APIs in the past, as nobody should've been using session cookies to authenticate their external API calls that weren't made by the application itself.

In any case, you can create API keys from the top right dropdown, on the page labeled "My API Keys". Each API key you create is only shown to you once, so you should keep a record of it somewhere. The API keys will have the same permissions you have in the system.

When attaching them to an API call, you can either submit them in the `X-API-Key` header or in the standardized `Authorization: Bearer your_api_key_here` format.
